deepchecks/nlp/text_data.py

```diff
@@ -313,11 +314,24 @@ def calculate_builtin_embeddings(self, model: str = 'miniLM', file_path: str = '
             'open_ai': using the ADA model in the open_ai library. Requires an API key.
         file_path : str, default: 'embeddings.npy'
             The path to save the embeddings to.
+        device : str, default: None
+            The device to use for calculating the embeddings. If None, the default device will be used.
+        long_doc_averaging : str, default 'warn'
+            How to handle long documents (docments that are longer than the model context window).
+            Can be either 'average+warn', 'average', 'truncate' or 'raise'. Averaging is done as described in
```

Comment:
alternatives to consider: 
1. always warn
2. default is raise and if average is selected dont warn

Reply:
1. Against - has to be an option to average without warning
2. I think you want things to "work" out of the box.

Reply:
(but without saying doc, see above comment)

Reply:
I think that though seperating is more "complete", it's simpler and easier to have just one parameter controlling everything about the behaviour for long docs. Also, the logic is that if you explicitly set a method, you don't need a warning for it ever.

Reply:
I really think this is too complicated.
And I get the warning, but Keep It Simple. I think this is a complicated solution that's not user friendly. Never seen a parameter that does 2 things like that, and that they are completely unrelated.